### DISCUSSION TITLE
Show environment variables in the command preview

### DISCUSSION THREAD

**Discussion by pioio** [go-task/task#1162] [Ideas]
When executing a command within the task, the task prints out the preview of command right before executing it.

In that preview, the vars are resolved to their values
`ssh {{.USER}}@{{.HOST}}`
becomes
`ssh root@foobar`

but env vars, are not resolved
`ssh ${USER}@${HOST}`
is printed as
`ssh ${USER}@${HOST}`

In some use cases (where the task's own env is controlled by an external application, e.g. a CI system) it would be useful to have the command preview resolve the env vars to their actual values before printing the command about to be execute. This would make
1) debugging output of the command easier (it's clear with which env the command was ran).
2) make it easier to copy/paste the preview, and rerun that specific command manually.

There's also valid usecases for using the current behaviour (where env vars are not being resolved). So, it would be useful to control in some way whether the env vars should be resolved in the preview or not, e.g. via a global, or a local to the task, bool property.

I see one complication here, environment vars can be defined in many different ways:
```
$FOO
${FOO}
${FOO:-BAR}
```

**Comment by rseichter** [comment]
For me, the lack of environment variable substitution is most painful when using the `--dry` or `--summary` flags. Consider this Taskfile:

```yaml
version: 3

env:
  FOO: bar

tasks:
  alpha:
    desc: Move to $FOO
    cmds:
      - "mv stuff $FOO"
```

Running `task alpha --dry`, I expect to see `task: [alpha] mv stuff bar`, but what I actually get is `mv stuff $FOO`. That's useless to me until I examine the Taskfile (in a few weeks I probably won't remember the default value for FOO).

In case you were wondering: Using a regular variable instead of an environment variable is not an option for my use case.
